### DISCUSSION TITLE
skywalking latency

### DISCUSSION THREAD

**Discussion by asdw123456wdsa** [apache/skywalking#11745] [Q&A]
Hello, I would like to inquire about the calculation method or logic for skywalking latency data?   Why can't data with a two-day delay be associated with a single day,
![截图_17048482585225](https://github.com/apache/skywalking/assets/156080732/ecf2e306-6321-4af8-9a1d-2a5ed4aba3ed)
![截图_17048483045958](https://github.com/apache/skywalking/assets/156080732/fe6965f8-328b-43bb-969d-f8ce5331185a)
![截图_17048483368875](https://github.com/apache/skywalking/assets/156080732/1c447532-27e0-45fd-bb5e-053d064d519d)

**Comment by wu-sheng** [comment]
One day means avg of 24 hours(24 records in hour level) latency data, two days mean avg of 2 days(2 records in day level). This is not from the minute level anymore.

**Reply by asdw123456wdsa** [reply]
thanks
Hourly level delay data and day data are calculated based on sampling
What about sampling frequency and sampling method
If not, please let me know

**Reply by wu-sheng** [reply]
Not sampling. This is a math problem.

Think in this way, if Day1 have 10k traffic with 10ms per request avg latency, and Days has 10 requests with 1000ms per request avg latency. If you do math for avg of Day1 and Day2, it is (10+1000)/2, which could be very high, and likely wrong. But that is a statistic match only.
If do the aggregation from the raw data, it would be very slow, and isn't worth.

**Reply by asdw123456wdsa** [reply]
According to the picture above
Is the algorithm for success rate the same as delay？
One day means avg of 24 hours(24 records in hour level) , two days mean avg of 2 days(2 records in day level).

**Reply by wu-sheng** [reply]
All metrics are in this way. It is a process for time series data, not for specific metrics.

**Reply by asdw123456wdsa** [reply]
Can you tell us what the algorithm is
Algorithm name or formula

**Reply by wu-sheng** [reply]
Isn't my description clear enough? An avg needs an algorithm? I am not following.

**Reply by asdw123456wdsa** [reply]
Is it like this

Calculate the total delay time of Day 1 and Day 2:
Total delay of Day 1: 10 milliseconds * 10,000 = 100,000 milliseconds
Total delay of Day 2: 1,000 milliseconds * 10 = 10,000 milliseconds
Calculate the total number of requests for both days:
Total number of requests: 10,000 + 10 = 10,100
Calculate the average delay:
Average delay: (100,000 + 10,000) / 10,100 ≈ 119.89 millisecond

**Reply by wu-sheng** [reply]
Don't guess. Check the data in your storage.
